src/GenericFactory/GenericFactory.sol

```diff
@@ -63,6 +63,8 @@ contract GenericFactory is MetaProxyDeployer {
     }
 
     constructor(address admin) {
+        if (admin == address(0)) revert E_BadAddress();
```

Comment:
What if you wanted to make a non-upgradeable factory?

Reply:
you can set the admin to 0xdead. But I'm not advocating for it. We can roll back if you want.

Reply:
Please change to comment "use address(0) to make it non-upgradeable"